### PR TITLE
Allow TiSpark retrieve row id

### PR DESCRIPTION
**Pull request by ilovesoup**
For implicit primary key, allow TiSpark to retrieve row_id. It's controlled by switch and if turned on, user will be able to see this column and retrieve it.

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tispark/367)

### CODE REVIEW

**Comment by Novemser** [comment]
/run-all-tests tikv=release-2.0 tidb=release-2.0 pd=release-2.0

**Comment by Novemser** [comment]
/run-all-tests tikv=release-2.0 tidb=release-2.0 pd=release-2.0

**Comment by Novemser** [comment]
/run-all-tests tikv=release-2.0 tidb=release-2.0 pd=release-2.0

File: tikv-client/src/main/java/com/pingcap/tikv/types/DataType.java [hunk 1]
```diff
@@ -63,7 +63,7 @@
   protected final MySQLType tp;
   // Not Encode/Decode flag, this is used to strict mysql type
   // such as not null, timestamp
-  protected final int flag;
+  protected int flag;
```
Comment (birdstorm) [inline tikv-client/src/main/java/com/pingcap/tikv/types/DataType.java]: final

File: core/src/main/scala/org/apache/spark/sql/TiContext.scala [hunk 2]
```diff
@@ -129,15 +129,20 @@ class TiContext(val session: SparkSession) extends Serializable with Logging {
...
-  // it just register table for later use
-  def tidbMapTable(dbName: String, tableName: String): DataFrame = {
+  def getDataFrame(dbName: String, tableName: String): DataFrame = {
     val tiRelation = new TiDBRelation(
       tiSession,
       new TiTableReference(dbName, tableName),
       meta
     )(sqlContext)
     val df = sqlContext.baseRelationToDataFrame(tiRelation)
+    df
```
[the remark this answers is not in the document]
Comment (ilovesoup) [reply core/src/main/scala/org/apache/spark/sql/TiContext.scala]: fixed

**Pull request merged**